### ISSUE TITLE
Character name carried over on M+ recordings

### ISSUE THREAD

**Issue by aza547** [aza547/wow-recorder#355]
![image](https://user-images.githubusercontent.com/32389325/229374963-fc0e853a-c7df-4626-8414-612063d03e89.png)

![image](https://user-images.githubusercontent.com/32389325/229374974-e4069e24-f452-450c-a678-2ba4cdc9750d.png)
[2023-04-01_22-22-30_-_Halls_of_Valor_23_0.txt](https://github.com/aza547/wow-recorder/files/11133150/2023-04-01_22-22-30_-_Halls_of_Valor_23_0.txt)
[WarcraftRecorder-2023-04-01.log](https://github.com/aza547/wow-recorder/files/11133152/WarcraftRecorder-2023-04-01.log)

**Comment by aza547** [comment]
Looks like the halls at 23:19 was failed, wiped on last boss and hearthed out at 23:03. 

At 23:12 player teleported to template and started a new run at 23:19 where we got the CHALLENGE_MODE_END event that always fires on new dungeon starting which triggered us to end the last run.  

My guess is that the character switch while a recording was still active caused the player to be updated when we saw a subsequent SPELL_AURA_APPLIED line with flags indicating the player, that failed to match with a combatant as the combatant was the character from the halls run (the priest, not the rogue), so we have no spec ID / class color. 

That' a pretty convoluted repo, but likely an easy fix is to make the player fields immutable (i.e. take the first SPELL_AURA_APPLIED event we see matching the player and never let it be overwritten).

Also another bug here is the +0 shown for an abandoned key.
